### PR TITLE
Simplify heartbeats + annotate alerts

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Changelist 

A few changes that should make driving less fragile.

1. Changed the heartbeat dependency graph. Now, board A only checks the heartbeat of board B if A uses a signal from board B. There currently are a few instances of boards checking other boards' heartbeats, but then not actually using signals from them. In my mind this is unnecessary and could only lead to unnecessary faults down the line. (check me on this!). This for example means the FSM checks no other boards heartbeats.

2. Moved various faults on the VC related to the LV battery and currents to warnings. If `AccumulatorRailOvercurrentFault` and `BatteryRailOvercurrentFault` is set, we'd probably see total loss of the LV bus, so no need to just exit drive state. If just one is set, there is no immediate reason to stop driving. If there is a `LvChargeFault` or `BoostControllerFault`, there is also no need to exit drive state (we'd probably only notice the consequences of the failure on the next power cycle). By making these faults instead of warnings we reduce the chances of spurious shutdowns.

3. Added descriptions for all alerts.

### Testing Done

Unit testing only.

### CODE REVIEW

File: can_bus/quadruna/RSM/RSM_alerts.json [hunk 1]
```diff
@@ -82,6 +82,10 @@
             "id": 452,
             "description": "VC Missing Heart beat"
         },
+        "MissingBMSHeartbeat": {
+            "id": 452,
+            "description": "BMS Missing Heart beat"
```
Comment (will-chaba) [inline can_bus/quadruna/RSM/RSM_alerts.json:87]: nit
```suggestion
            "description": "BMS Missing Heartbeat"
```

File: firmware/quadruna/FSM/src/app/app_mainState.c [hunk 2]
```diff
@@ -11,30 +11,18 @@
 #include "app_suspension.h"
 #include "app_loadCell.h"
 #include "app_shdnLoop.h"
-#include "app_heartbeatMonitor.h"
 
 void mainStateRunOnTick100Hz(void)
 {
+    app_canTx_FSM_Heartbeat_set(true);
```
Comment (will-chaba) [inline firmware/quadruna/FSM/src/app/app_mainState.c:17]: I'm a bit confused what's going on here. Do any boards consume the FSM heartbeat? if not why not remove the messages altogether?

File: firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h [hunk 3]
```diff
@@ -110,14 +110,14 @@ class VcBaseStateMachineTest : public BaseStateMachineTest
     // config for heartbeat monitor (can funcs and flags)
     // VC relies on FSM, RSM, BMS, CRIT
```
Comment (will-chaba) [inline firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h:111]: Update this

File: firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h [hunk 4]
```diff
@@ -110,14 +110,14 @@ class VcBaseStateMachineTest : public BaseStateMachineTest
...
     bool heartbeatMonitorChecklist[HEARTBEAT_BOARD_COUNT] = {
-        [BMS_HEARTBEAT_BOARD] = true, [VC_HEARTBEAT_BOARD] = false,  [RSM_HEARTBEAT_BOARD] = true,
+        [BMS_HEARTBEAT_BOARD] = true, [VC_HEARTBEAT_BOARD] = false,  [RSM_HEARTBEAT_BOARD] = false,
```
Comment (will-chaba) [inline firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h:113]: It was my understanding that the RSM was required as it had the coolant sensors to check for over-heating conditions, and if it drops off we may not shutdown due to overtemp condition

**Review by will-chaba — COMMENTED** [review]
Looks good, but think the VC should still depend on RSM to be able to shutdown due to over-heating

File: firmware/quadruna/FSM/src/app/app_mainState.c (hunk 2 above)
Comment (gtaharaedmonds) [reply firmware/quadruna/FSM/src/app/app_mainState.c:17]: Yes, the VC does for accelerator pedals and brakes.

File: firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h (hunk 3 above)
Comment (gtaharaedmonds) [reply firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h:111]: this is still true

File: firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h (hunk 4 above)
Comment (gtaharaedmonds) [reply firmware/quadruna/VC/test/test_vcBaseStateMachineTest.h:113]: good call, will fix this.

**Review by will-chaba — APPROVED** [review]

**Pull request merged**